From 07c8cbc636fe81fbc1f61d68400e16b0db313783 Mon Sep 17 00:00:00 2001
From: Rameshkumar Sundaram <quic_ramess@quicinc.com>
Date: Tue, 28 Feb 2023 00:43:54 +0530
Subject: [PATCH 4/4] wifi : ath11k : fetch mbssid tx vif from deflink

Make changes to fetch mbssid tx vif from vifs->deflink
which is restructured as part of MLO MBSSID Support.

Signed-off-by: Rameshkumar Sundaram <quic_ramess@quicinc.com>
---
 drivers/net/wireless/ath/ath11k/mac.c | 26 +++++++++++++-------------
 1 file changed, 13 insertions(+), 13 deletions(-)

--- a/drivers/net/wireless/ath/ath11k/mac.c
+++ b/drivers/net/wireless/ath/ath11k/mac.c
@@ -1763,10 +1763,10 @@ static int ath11k_mac_setup_bcn_tmpl_ema
 	int ret = 0;
 	u8 i = 0;
 
-	if (!arvif->vif->mbssid_tx_vif)
+	if (!arvif->vif->bss_conf.mbssid_tx_vif)
 		return -1;
 
-	tx_arvif = ath11k_vif_to_arvif(arvif->vif->mbssid_tx_vif);
+	tx_arvif = (void *)arvif->vif->bss_conf.mbssid_tx_vif->drv_priv;
 
 	beacons = ieee80211_beacon_get_template_ema_list(tx_arvif->ar->hw,
 							 tx_arvif->vif, 0);
@@ -1818,8 +1818,8 @@ static int ath11k_mac_setup_bcn_tmpl_mbs
 	struct sk_buff *bcn;
 	int ret;
 
-	if (vif->mbssid_tx_vif) {
-		tx_arvif = ath11k_vif_to_arvif(vif->mbssid_tx_vif);
+	if (arvif->vif->bss_conf.mbssid_tx_vif) {
+		tx_arvif = (void *)arvif->vif->bss_conf.mbssid_tx_vif->drv_priv;
 		if (tx_arvif != arvif) {
 			ar = tx_arvif->ar;
 			ab = ar->ab;
@@ -1899,8 +1899,8 @@ static void ath11k_control_beaconing(str
 	int ret = 0;
 	bool found_vdev = false;
 
-	if (arvif->vif->mbssid_tx_vif)
-		tx_arvif = (void *)arvif->vif->mbssid_tx_vif->drv_priv;
+	if (arvif->vif->bss_conf.mbssid_tx_vif)
+		tx_arvif = (void *)arvif->vif->bss_conf.mbssid_tx_vif->drv_priv;
 
 	lockdep_assert_held(&arvif->ar->conf_mutex);
 
@@ -3502,9 +3502,8 @@ static void ath11k_bss_disassoc(struct i
 			    arvif->vdev_id, ret);
 
 	arvif->is_up = false;
-
-	if (arvif->vif->mbssid_tx_vif) {
-		tx_arvif = (void *)arvif->vif->mbssid_tx_vif->drv_priv;
+	if (arvif->vif->bss_conf.mbssid_tx_vif) {
+		tx_arvif = (void *)arvif->vif->bss_conf.mbssid_tx_vif->drv_priv;
 		if (tx_arvif != arvif)
 			tx_arvif->nontransmitting_vif_count--;
 	}
@@ -8179,7 +8178,7 @@ static int ath11k_mac_setup_vdev_params_
 	struct ieee80211_vif *tx_vif;
 
 	*tx_vdev_id = 0;
-	tx_vif = arvif->vif->mbssid_tx_vif;
+	tx_vif = arvif->vif->bss_conf.mbssid_tx_vif;
 	if (!tx_vif) {
 		*flags = WMI_HOST_VDEV_FLAGS_NON_MBSSID_AP;
 		return 0;
@@ -9506,9 +9505,8 @@ ath11k_mac_update_vif_chan(struct ath11k
 		params.vdev_id = arvif->vdev_id;
 		params.aid = arvif->aid;
 		params.bssid = arvif->bssid;
-
-		if (arvif->vif->mbssid_tx_vif) {
-			tx_arvif = (void *)arvif->vif->mbssid_tx_vif->drv_priv;
+		if (arvif->vif->bss_conf.mbssid_tx_vif) {
+			tx_arvif = (void *)arvif->vif->bss_conf.mbssid_tx_vif->drv_priv;
 			params.tx_bssid = tx_arvif->bssid;
 			params.profile_idx = arvif->vif->bss_conf.bssid_index;
 			params.profile_count = tx_arvif->nontransmitting_vif_count;
